### ISSUE TITLE
Can't be installed on a raspberry pi

### ISSUE THREAD

**Issue by fdev31** [johannesnoordanus/grblhud#1]
While very dependency can be installed flawlessly, getting grblhud running is a pain. The main reasons are:

- package metadata on pypi is wrong, it can run with a much older Python version
- there is no way to quickly fix it and install on a virtual environment (you need to understand how it should run, nothing describes it)
- recommended way using flit doesn't work (build error on raspberry pi)

**Comment by johannesnoordanus** [comment]
Ok, I understand raspberry pi has a (much) older python version?

**Comment by fdev31** [comment]
It's a 3.7 here with an up to date system.  It's fully compatible with your code (I get the exact same errors on python 3.11).

**Comment by johannesnoordanus** [comment]
> It's a 3.7 here with an up to date system. It's fully compatible with your code (I get the exact same errors on python 3.11).

Ok thanks, I will lower the requirement and publish again.(tomorrow)

**Comment by johannesnoordanus** [comment]
> The main reasons are:
> 
> * package metadata on pypi is wrong, it can run with a much older Python version

Ok, I will correct that.
> * there is no way to quickly fix it and install on a virtual environment (you need to understand how it should run, nothing describes it)

I guess Flit isn’t intended for that. I think the way python packages are made to run is default though.
> * recommended way using flit doesn't work (build error on raspberry pi)

Ok, but what error? And did ‘flit’ itself install?

**Comment by fdev31** [comment]
I'm traveling a few weeks and don't have access to my raspberry pi anymore during that time. But that would be just simpler to provide a pyprojects file... adding an unnecessary dependency to your project can only add complications...

**Comment by fdev31** [comment]
Also the output is a huge issue on some systems, I tested on a gaming computer and it's quite better but on something light using "big" images it's not usable, on a pi for instance, or via a slow ssh I guess.
Also it would be good (also because apparently there are some options for problems there) to just type "run" to run the latest loaded file...

**Comment by johannesnoordanus** [comment]
> I'm traveling a few weeks and don't have access to my raspberry pi anymore during that time. But that would be just simpler to provide a pyprojects file... adding an unnecessary dependency to your project can only add complications...

I agree, but the python version is a concern, I encountered a lot of compatibility problems when python versions are too old.
That is certainly true for version 2.xxx to 3.xxx. So for the time being I will lower the version requirement.

I do not have experience with pyproject files, and would like to keep things a simple as possible. I will look at that in the future, but I cannot do all things at once. 
Like you I guess, I have several projects ongoing for which I need this software, and that's the reason I am developing this, but I have severe time constraints.

**Comment by johannesnoordanus** [comment]
> Also the output is a huge issue on some systems, I tested on a gaming computer and it's quite better but on something light using "big" images it's not usable, on a pi for instance, or via a slow ssh I guess.
> Also it would be good (also because apparently there are some options for problems there) to just type "run" to run the latest loaded file...

I agree, I'll cap that off and show only the first and last 'n' lines. I'll also do the load in parallel so the main loop will keep control and commands like 'exit' etc. will work. I might add a command to break off loading files. Also loading is't always necessary, it is there to make things like running self defined loops (to repeat a cut for example) manageable. So I will add a direct run method, which is called 'run' and make a run from the buffer optional.
Anyway, I know how to fix the issues you mentioned. And I will do that.

**Comment by johannesnoordanus** [comment]
Updated the python requirement.
I would like to close this issue.

**Comment by fdev31** [comment]
Since I'm not able to re-test soon, I'll close it now and eventually re-open in case something was missed, later when I can run tests.

**Comment by johannesnoordanus** [comment]
> Since I'm not able to re-test soon, I'll close it now and eventually re-open in case something was missed, later when I can run tests.

Ok, it is grblhud 1.6.0 (and higher) that has the updates (not 1.5.2).

**Comment by johannesnoordanus** [comment]
New issue by fdev31: "I see grblhud requires scikit-images because of the gcode2image dependency... I think it would be great to have it optional for people using it in command line only (typical use case is like mine: a raspberry pi is plugged to the machine and is used as a server). At the moment it makes the installation fail on a raspberry because of the required scikit-image version (not even talking about the crazy amount of time to install this + numpy from pypi)."

**Comment by johannesnoordanus** [comment]
Resolved this in grblhud version 1.6.1.

**Comment by johannesnoordanus** [comment]
Use latest version 1.7.3. This is fixed.

**Comment by johannesnoordanus** [comment]
Fixed. Use latest version 1.7.3.
